rmw_cyclonedds_cpp/src/rmw_node.cpp

```diff
@@ -675,45 +675,75 @@ char * get_security_file_URI(
   return ret;
 }
 
-void store_security_filepath_in_qos(
+bool store_security_filepath_in_qos(
   dds_qos_t * qos, const char * qos_property_name, const char * file_name,
   const rmw_node_security_options_t * security_options)
 {
+  bool ret = false;
   rcutils_allocator_t allocator = rcutils_get_default_allocator();
 
-  char * security_file_path = get_security_file_URI(
-    file_name, security_options->security_root_path, allocator);
-  if (security_file_path != nullptr) {
-    dds_qset_prop(qos, qos_property_name, security_file_path);
-    allocator.deallocate(security_file_path, allocator.state);
+  if (security_options->security_root_path != nullptr) {
+    char * security_file_path = get_security_file_URI(
+      file_name, security_options->security_root_path, allocator);
+    if (security_file_path != nullptr) {
+      dds_qset_prop(qos, qos_property_name, security_file_path);
+      ret = true;
+      allocator.deallocate(security_file_path, allocator.state);
+    }
   }
+  return ret;
+}
+
+/* Disable cyclone security */
+rmw_ret_t unconfigure_qos_for_security(dds_qos_t * qos)
+{
+  dds_qunset_prop(qos, "dds.sec.auth.identity_ca");
```

Comment:
This is very error-prone. If we ever add support for something new we'll need to make sure we also remember to unset it here. It would be better to stage the QoS changes (e.g. in a struct like we discussed initially) and make sure it will succeed before continuing as opposed to making the changes and then reverting them.